internal/cnpi/plugin/client/lifecycle.go

```diff
@@ -53,15 +49,20 @@ func (data *data) LifecycleHook(
 		return nil, err
 	}
 	gvk := object.GetObjectKind().GroupVersionKind()
-	if gvk.Kind == "" || gvk.Version == "" {
-		gvk, err = apiutil.GVKForObject(object, runtimeScheme)
+	if gvk.Empty() {
+		if schema == nil {
+			contextLogger.Info("missing scheme, unable to infer gvk for object", "object", object)
+			return object, nil
+		}
+		gvk, err = apiutil.GVKForObject(object, schema)
 		if err != nil {
-			contextLogger.Trace("skipping unknown object", "object", object)
+			contextLogger.Info("missing gvk for object, got error trying to infer it from schema",
+				"object", object, "error", err)
 			// Skip unknown object
-			return nil, nil
+			return object, nil
```

Comment:
this is already covered by https://github.com/cloudnative-pg/cloudnative-pg/pull/4282, I do think it's the right thing to do.